src/core/analytics/visibilityanalyticshandler.js

```diff
@@ -1,34 +1,42 @@
 import { isIE } from '../utils/useragent';
 import AnalyticsEvent from '../analytics/analyticsevent';
 import Searcher from '../models/searcher';
+import StorageKeys from '../storage/storagekeys';
+
+const RESULTS_VISIBILITY_EVENT = {
+  HIDDEN: 'RESULTS_HIDDEN',
+  UNHIDDEN: 'RESULTS_UNHIDDEN'
+};
 
 /**
  * Manages the document's visibility status and handles any visibility related analytics events.
  */
 export default class VisibilityAnalyticsHandler {
   constructor (analyticsReporterService, verticalKey) {
-    this._documentVisibilityState = undefined;
+    this._previousResultsVisibilityEvent = undefined;
     this._analyticsReporterService = analyticsReporterService;
     this._verticalKey = verticalKey;
   }
 
   /**
    * Initialize visibility change event listener(s) to send analytics events
    * when a result page have become visible or have been hidden.
+   *
+   * @param {Storage} storage - a container around application state
    */
-  initVisibilityChangeListeners () {
+  initVisibilityChangeListeners (storage) {
     /**
      * Safari desktop listener and IE11 listeners fire visibility change event twice when switch
-     * to new tab and then close browser. This variable is used to ensure RESULTS_HIDDEN analytics event
-     * does not get send again if the page is already hidden.
+     * to new tab and then close browser. Variable "_previousResultsVisibilityEvent" is used to ensure
+     * RESULTS_HIDDEN analytics event does not get send again if the page is already hidden.
      */
     document.addEventListener('visibilitychange', () => {
-      if (document.visibilityState === 'hidden' && this._documentVisibilityState !== 'hidden') {
-        this._documentVisibilityState = 'hidden';
-        this._reportVisibilityChangeEvent('RESULTS_HIDDEN');
-      } else if (document.visibilityState === 'visible') {
-        this._documentVisibilityState = 'visible';
-        this._reportVisibilityChangeEvent('RESULTS_UNHIDDEN');
+      if (document.visibilityState === 'hidden' && this._previousResultsVisibilityEvent !== RESULTS_VISIBILITY_EVENT.HIDDEN) {
```

Comment:
Is there some way we could make these conditions a little more readable? I was thinking maybe some kind of helper method so we can give it a proper name, otherwise as somebody not familiar for the code it's hard to grasp exactly what is going on and why

Reply:
sure, I updated to separate some of these conditions in if statements to its own helper method. Hopefully that's clearer